app/components/lesson_new_position_component.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+class LessonNewPositionComponent < ViewComponent::Base
+  attr_reader :lesson, :form
+
+  def initialize(lesson:, form:)
+    @lesson = lesson
+    @form = form
+  end
+
+  def collection
+    array = (lesson.course_module.course_lessons - [lesson]).each_with_index.map do |l, index|
+      OpenStruct.new(title: l.title, position: index + 2)
+    end
+
+    array.compact!
+
+    array.prepend(OpenStruct.new(title: "Set as first lesson", position: 1))
+
+    array
+  end
+
+  def selected
+    lesson.position
+  end
+
+  def render?
+    lesson.persisted?
+  end
```

Comment:
> So will that not render when we create new lesson? Is that because we don't know the module, so we would have a hard time showing the right lessons to pick?

Yes

> Maybe a better solution would be to manage it in the view - either have an if statement in shared fields

But isn't that what we're doing here? The component knows when to render itself and not the caller. Also we have one fewer explicit `if` statement to comprehend.

> or maybe add this field to the edit view and not to the new view?

These are shared now via a partial so we'd have to duplicate everything

Reply:
done